gremlin-go/driver/connection_test.go

```diff
@@ -52,6 +53,49 @@ const basicAuthWithSsl = "wss://localhost:45941/gremlin"
 
 var testNames = []string{"Lyndon", "Yang", "Simon", "Rithin", "Alexey", "Valentyn"}
 
+func init() {
+	count := 0
+	// Test server status before starting the integration tests
+	log.Println("Initializing connection tests, testing no auth and auth server connection status.")
+	for {
+		conn, err := NewDriverRemoteConnection(getEnvOrDefaultString("GREMLIN_SERVER_URL", noAuthUrl))
+		if err == nil {
+			log.Printf("Connection %v successful, no auth server initialized.", conn)
+			conn.Close()
+			break
+		}
+		log.Println("Connection unsuccessful, trying again in 1 second.")
+		time.Sleep(1 * time.Second)
+		count++
+		if count%30 == 0 {
+			log.Println("===============================================================================")
+			log.Printf("Connection unsuccessful after %d tries, you may wish to check server status and rerun the test suite", count)
+			log.Println("===============================================================================")
+		}
+	}
+	count = 0
+	for {
+		conn, err := NewDriverRemoteConnection(getEnvOrDefaultString("GREMLIN_SERVER_BASIC_AUTH_URL", basicAuthWithSsl),
+			func(settings *DriverRemoteConnectionSettings) {
+				settings.TlsConfig = &tls.Config{InsecureSkipVerify: true}
+				settings.AuthInfo = getBasicAuthInfo()
+			})
+		if err == nil {
+			log.Printf("Connection %v successful, auth server initialized.", conn)
+			conn.Close()
+			break
+		}
+		log.Println("Connection unsuccessful, trying again in 1 second.")
+		time.Sleep(1 * time.Second)
+		count++
+		if count%30 == 0 {
+			log.Println("===============================================================================")
+			log.Printf("Connection unsuccessful after %d tries, you may wish to check server status and rerun the test suite", count)
+			log.Println("===============================================================================")
+		}
+	}
+}
+
```

Comment:
We need to figure out a way to do this in docker.

Reply:
Done, see latest commit.